components/list/list-item-drag-mixin.js

```diff
@@ -33,86 +95,299 @@ export const ListItemDragMixin = superclass => class extends superclass {
+		dropSpots.updateTime(e.timeStamp);
+		e.preventDefault();
+	}
+
+	_dragStart(e) {
+		e.dataTransfer.setData('text/plain', `${this.dragText}`);
+		e.dataTransfer.effectAllowed = 'move';
+
+		const nodeImage = this.shadowRoot.querySelector('.d2l-list-item-drag-image') || this;
+		e.dataTransfer.setDragImage(nodeImage, 50, 50);
+
+		dropSpotsFactory(this);
+
+		setTimeout(() => {
+			this.beingDragged = true;
+		});
+	}
+
+	_dragTopEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		const dropSpots = dropSpotsFactory();
+		dropSpots.setDestination(this, move.above);
+		this._inTopArea = true;
+	}
+
+	_dragUpperEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		if (this._inTopArea) {
+			const dropSpots = dropSpotsFactory();
+			dropSpots.setDestination(this, move.below);
+			this._inTopArea = false;
+		}
+	}
+
+	_drop() {
+		const dropSpots = dropSpotsFactory();
+		dropSpots.setDestination(this, dropSpots.moveBeforeDestination);
+	}
+
+	_hostDragEnter(e) {
+		const dropSpots = dropSpotsFactory();
+		if (this === dropSpots.target) {
+			return;
+		}
+		dropSpots.addVisitor(this);
+		this._dropTarget = true;
+		e.dataTransfer.dropEffect = 'move';
+	}
+
+	_moveItem(targetKey, destinationKey, moveBeforeDestination = false, temporaryMovement = false) {
 		this.dispatchEvent(new CustomEvent('d2l-list-item-position', {
-			detail: detail,
+			detail: new NewPositionEventDetails({targetKey, destinationKey, moveBeforeDestination, temporaryMovement}),
 			bubbles: true
 		}));
 	}
 
-	_dragEnter() {
-		this._dragStartHandler();
+	_renderBottomPlacementMarker(renderTemplate) {
+		return this._bottomPlacementMarker ? html`<div class="d2l-list-item-drag-bottom-marker">${renderTemplate}</div>` : null;
+	}
+
+	_renderDraggableArea(templateMethod) {
+		templateMethod = templateMethod || (dragArea => dragArea);
+		return this.draggable && !this._keyboardMode ? templateMethod(html`
+			<div
+				class="d2l-list-item-drag-area"
+				draggable="true"
+				@click="${this._dragAreaClick}"
+				@dragstart="${this._dragStart}"
+				@dragend="${this._dragEnd}"
+				>
+			</div>
+		`) : nothing;
+	}
+
+	_renderDragHandle(templateMethod) {
+		templateMethod = templateMethod || (dragHandle => dragHandle);
+		return this.draggable ? templateMethod(html`
+			<d2l-list-item-drag-handle id="${this._itemDragId}" @d2l-list-item-drag-handle-action="${this._dragHandleActions}"></d2l-list-item-drag-handle>
+		`) : nothing;
 	}
 
-	_dragExit() {
-		this._dispatchDragEvent(null);
+	_renderDropArea(templateMethod) {
+		templateMethod = templateMethod || (dropArea => dropArea);
+		return this.draggable && this._dropTarget ? templateMethod(html`
+			<div class="d2l-list-item-drag-drop-grid" @drop="${this._drop}" @dragover="${this._dragOver}">
+				<div @dragenter="${this._dragTopEnter}"></div>
+				<div @dragenter="${this._dragUpperEnter}"></div>
+				<div @dragenter="${this._dragLowerEnter}"></div>
+				<div @dragenter="${this._dragBottomEnter}"></div>
+			</div>
```

Comment:
Defines: https://desire2learn.atlassian.net/wiki/spaces/POL/pages/996083189/US114586+wc-list+Refactor+for+Drag+n+Drop#Drag-Areas-for-a-List-Item